src/order/boolean_algebra.lean

```diff
@@ -923,3 +923,29 @@ protected def function.injective.boolean_algebra [has_sup α] [has_inf α] [has_
   ..hf.generalized_boolean_algebra f map_sup map_inf map_bot map_sdiff }
 
 end lift
+
+namespace punit
+variables (a b : punit.{u+1})
+
+instance : boolean_algebra punit :=
```

Comment:
I agree with moving these instances out of `punit_instances`, but I think it might be best doing it in a separate PR.

Reply:
Yeah I'm opening it right now.